src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3377,6 +3377,29 @@ TransactionStateMachine(WorkerSession *session)
 			case REMOTE_TRANS_SENT_COMMAND:
 			{
 				TaskPlacementExecution *placementExecution = session->currentTask;
+				if (placementExecution == NULL)
+				{
+					/*
+					 * We have seen accounts in production where the placementExecution
```

Comment:
I don't have a strong opinion on this, but I would probably not explain in this way, but more around:

```
Due to some integration problems with failure handling of SAVEPOINTs, it is sometimes possible that the transaction state remains REMOTE_TRANS_SENT_COMMAND from the previous command. In such cases, the placement execution would not have been set yet. Until we improve the integration, we bail out early here as the internal state is not consistent.
```